### PR TITLE
maryia/DTRA-260/TS migration of /Constants files & SmartChart/Helpers files in Trader

### PR DESCRIPTION
**Pull request by maryia-deriv**
## Changes:

- migrate validation-rules to ts
- remove 2 unused ui.js files
- migrate index.js to ts
- remove unused markers.js
- migrate Constants/barriers.js file to ts
- migrate Helpers/barriers.js file and its test to ts
- migrate chart-barrier-store.js to ts
- remove unused chart-marker-store.js

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

**Review by kate-deriv — APPROVED** [review]

File: packages/trader/src/Stores/Modules/SmartChart/Helpers/barriers.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import type { ChartBarrierStore } from '../chart-barrier-store';
+
+export type TBarrier = ChartBarrierStore & { key?: string };
```
Comment (henry-deriv) [inline packages/trader/src/Stores/Modules/SmartChart/Helpers/barriers.ts:3]: for me i ended up adding key as a prop in chartbarrierstore in my PR

File: packages/trader/src/Stores/Modules/Trading/Constants/validation-rules.ts [hunk 2]
```diff
@@ -0,0 +1,241 @@
+import { localize } from '@deriv/translations';
+import { isHourValid, isMinuteValid, isTimeValid, toMoment } from '@deriv/shared';
+import { isSessionAvailable } from '../Helpers/start-date';
+import { TTradeStore } from 'Types';
+
+type TOptions = {
+    min?: number;
```
Comment (henry-deriv) [inline packages/trader/src/Stores/Modules/Trading/Constants/validation-rules.ts:7]: why not number | string as well?

**Review by henry-deriv — COMMENTED** [review]
Left comments

File: packages/trader/src/Stores/Modules/SmartChart/Helpers/barriers.ts (hunk 1 above)
Comment (maryia-deriv) [reply packages/trader/src/Stores/Modules/SmartChart/Helpers/barriers.ts:3]: @henry-deriv the thing is that ChartBarrierStore has nothing to do with the key, the key is assigned afterwards to a resulting object which is returned as a result of `new ChartBarrierStore(...)` call.

Comment (henry-deriv) [reply packages/trader/src/Stores/Modules/SmartChart/Helpers/barriers.ts:3]: I made the same change in my PR

Comment (maryia-deriv) [reply packages/trader/src/Stores/Modules/SmartChart/Helpers/barriers.ts:3]: thank you🙏🏻

File: packages/trader/src/Stores/Modules/Trading/Constants/validation-rules.ts (hunk 2 above)
Comment (maryia-deriv) [reply packages/trader/src/Stores/Modules/Trading/Constants/validation-rules.ts]: good point👍🏻 reused existing TRuleOptions type from validator.ts :)

**Review by henry-deriv — APPROVED** [review]

**Review by kate-deriv — APPROVED** [review]

**Pull request merged**